src/startupscriptgenerator/src/common/parseUserCommands.go

```diff
@@ -0,0 +1,56 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+package common
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+func ParseUserRunCommand(sourcePath string) string {
+
+	appsvcFile, err := os.Open(sourcePath)
+	if err != nil {
+		return ""
+	}
+	defer appsvcFile.Close()
+
+	const runHeading string = "run:" // format of run command- "run: gunicorn myapp.app --workers 5"
+	runCommand := ""
+	var isRunCommandFound bool
+	scanner := bufio.NewScanner(appsvcFile)
+
+	for scanner.Scan() {
+		//fmt.Println(scanner.Text())
+		indexOfRunHeading := strings.Index(scanner.Text(), runHeading)
+
+		if indexOfRunHeading < 0 && strings.Contains(scanner.Text(), ":") && isRunCommandFound {
+			// runCommand already found
+			// not considering any other customized commands
+			break
+		}
+
+		if isRunCommandFound || (indexOfRunHeading >= 0 && len(scanner.Text()) > len(runHeading)) {
```

Comment:
Would you mind clarifying what's going on in this block? Are we checking for a multiline run command that the user provided in the `appsvc.yaml` file?

Reply:
Could we increase readability by moving logic into a variable? If you also move the https://github.com/microsoft/Oryx/pull/1430/files#r932680872 into a var you can do something like:
```
foo = ...
bar = ...
if foo & bar {
    // do stuff
}
```

Reply:
@cormacpayne: it checks multiple scenarios:

1) this validifies only one run command-block with a heading `run:`
2) it checks if the `appsvc.yaml` contains the heading-string `run:`. then, it considers the substring as the run-command followed by the `run:` heading.3) according to the example of a standard appsvc.yaml file(https://github.com/JasonFreeberg/appsvc.yaml-examples) I considered any command-block should have a heading with a colon `:`. E.g. `run:`, `build:`, etc.
here, I did not consider any further command-block when the run command-block is already found.
4)  a multi lined run-command block is considered as a valid run command-block.